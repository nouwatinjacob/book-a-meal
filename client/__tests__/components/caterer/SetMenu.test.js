import React from 'react';
import {
  shallow
} from 'enzyme';
import sweetalert from 'sweetalert';
import {
  SetMenu,
  mapStateToProps
} from '../../../src/components/caterer/SetMenu.jsx';
import {
  allMealResponse
} from '../../__mockData__/mockMeal';

const props = {
  getMeals: jest.fn(),
  setMenuAction: () => Promise.resolve(),
  mealState: {
    paginate: allMealResponse.paginate,
  }
};

jest.mock('sweetalert');

describe('SetMenu component', () => {
  const setup = () => shallow(<SetMenu {...props} />);

  it('should render SetMenu component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'onDateChange'
  when date is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        value: '2018-05-09'
      }
    };

    const onDateChangeSpy = jest.spyOn(
      wrapper.instance(),
      'onDateChange'
    );
    wrapper.instance().onDateChange(event);
    expect(onDateChangeSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onCheck'
  when date is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        value: 1
      }
    };

    const onCheckSpy = jest.spyOn(
      wrapper.instance(),
      'onCheck'
    );
    wrapper.instance().onCheck(event);
    expect(onCheckSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onFormSubmit'
  when modify order button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    wrapper.setState({
      menuData: {
        menuDate: '2018-12-09',
        mealId: [2, 3]
      }
    });

    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'onFormSubmit'
    );
    sweetalert.mockResolvedValue(Promise.resolve(true));
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handlePageClick'
  if pagination button is clicked`, (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePageClickSpy = jest.spyOn(
      wrapper.instance(),
      'handlePageClick'
    );
    wrapper.instance().handlePageClick(data);
    expect(handlePageClickSpy).toHaveBeenCalled();

    done();
  });
});