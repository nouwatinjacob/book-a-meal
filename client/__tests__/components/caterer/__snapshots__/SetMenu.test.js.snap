// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetMenu component should render SetMenu component correctly 1`] = `
<div
  className="container"
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <CatererHeader />
  <div
    className="wrapper"
  >
    <div
      className="login"
    >
      <div
        className="login-form"
      >
        <h3>
          Add New Menu
        </h3>
        <form
          onSubmit={[Function]}
        >
          <input
            id="my-input"
            min="2018-09-14"
            name="date"
            onChange={[Function]}
            type="date"
          />
          <br />
          <h6>
            Pick meals to be added to your Menu
          </h6>
          <button
            className="button warning"
            id="long-button"
          >
            Save
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
